### ISSUE TITLE
Teach yesql about params vs escaped vars

### ISSUE THREAD

**Issue by emidln** [krisajenkins/yesql#87]
Yesql should provide a method to process escaped variables as well as params. This comes up if you want to do something relatively benign such as:

```sql
-- name: foo-ordered
SELECT * FROM foo f ORDER BY f.ts :direction;
```

Currently, yesql treats :direction as a param, which results in this jdbc arrray:

```clj
; (defqueries "/path/to/foo.sql")
; (foo-ordered "DESC") 
["SELECT * FROM foo f ORDER BY f.ts ?" "DESC"]
```

In the ideal situation, we'd end up with:

```clj
; (defqueries "/path/to/foo.sql" (fn escape [x] (identity x))
; (foo-order "DESC")
["SELECT * FROM foo f ORDER BY f.ts DESC"]
```

It would be very useful to provide an alternate mechanism to signal quoting. Additional grammar utilizing something like :: (bikeshed here) as a prefix for quote would make it easy to allow users to provide an escaping function to `defqueries` to handle escaping before interpolation.

**Comment by AngryGami** [comment]
https://github.com/krisajenkins/yesql/issues/94

**Comment by ricardojmendez** [comment]
Agreed.  Or even something like converting Clojure symbols to their direct, unescaped string value, so that one could just pass `(foo-ordered 'DESC)`.

**Comment by dan-silva** [comment]
+1 its the only thing blocking me from using this library :/

**Comment by russellwhitaker** [comment]
:+1:  I was using yesql happily until I just ran into the need for this, and the lack of this facility now hurts.

**Comment by AngryGami** [comment]
https://github.com/AngryGami/yesql

**Comment by gberenfield** [comment]
At this point this solves it for me:
https://github.com/layerware/hugsql

**Comment by russellwhitaker** [comment]
@AngryGami thanks for the pointer to your fork, interesting. @gberenfield yes, last night I found hugsql and was very impressed, and will be checking it out on the next round of work on my project, thanks.
